src/com/maddyhome/idea/vim/group/MotionGroup.java

```diff
@@ -1215,21 +1230,38 @@ public int moveCaretToLineStartOffset(@NotNull Editor editor) {
   }
 
   public int moveCaretToLineScreenStart(@NotNull Editor editor) {
-    // TODO: Add multiple carets support
-    int col = EditorHelper.getVisualColumnAtLeftOfScreen(editor);
-    return moveCaretToColumn(editor, editor.getCaretModel().getPrimaryCaret(), col, false);
+    return moveCaretToLineScreenStart(editor, editor.getCaretModel().getPrimaryCaret());
+  }
+
+  public int moveCaretToLineScreenStart(@NotNull Editor editor, @NotNull Caret caret) {
+    final int col = getLeftmostScreenColumn(editor);
+    return moveCaretToColumn(editor, caret, col, false);
+  }
+
+  private int getLeftmostScreenColumn(@NotNull Editor editor) {
+    final int leftmost = EditorHelper.getVisualColumnAtLeftOfScreen(editor);
+    return leftmost == 0 ?
```

Comment:
It looks like this fix is actually applicable to all usages of `getVisualColumnAtLeftOfScreen()`. Also it's unrelated to the problem of multiple carets. Let's move it into a separate PR and check the other usages of `getVisualColumnAtLeftOfScreen()`.